### PR TITLE
[New] `no-unused-modules`: Add `ignoreUnusedTypeExports` option

### PR DESCRIPTION
**Pull request by silverwind**
Fixes: https://github.com/import-js/eslint-plugin-import/issues/2694

### CODE REVIEW

**Comment by silverwind** [comment]
Will tweak tests a bit.

**Comment by ljharb** [comment]
This seems great, except that it's best when boolean arguments default to false. Can you come up with an alternative name for it so it can default to false?

**Comment by silverwind** [comment]
> This seems great, except that it's best when boolean arguments default to false. Can you come up with an alternative name for it so it can default to false?

Hmm I could invert the name and function but thought that the current name matches nicely into the existing options `missingExports` and `unusedExports`.

**Comment by silverwind** [comment]
Inverted it, so now it's `ignoreUnusedTypeExports` defaulting to `false`.

**Comment by ljharb** [comment]
Your original option name definitely did match better, but i prioritize booleans defaulting to false over option names :-) thanks!

File: docs/rules/no-unused-modules.md [hunk 1]
```diff
@@ -29,6 +29,7 @@ This rule takes the following option:
 
  - **`missingExports`**: if `true`, files without any exports are reported (defaults to `false`)
  - **`unusedExports`**: if `true`, exports without any static usage within other modules are reported (defaults to `false`)
+ - **`ignoreUnusedTypeExports`**: if `true`, type exports without any static usage within other modules are reported (defaults to `false` and depends on `unusedExports` being `true`)
```
Comment (silverwind) [inline docs/rules/no-unused-modules.md:32]:
```suggestion
 - **`ignoreUnusedTypeExports`**: if `true`, type exports without any static usage within other modules are reported (defaults to `false` and has no effect unless `unusedExports` is `true`)
```

**Comment by silverwind** [comment]
This CI failures seem intermittent and the coverage annotation "Added line #L518 was not covered by tests" is incorrect, I verified the line is being hit during tests.

**Comment by ljharb** [comment]
node 22.5 itself is broken rn, so i'll rerun those failing jobs once a fixed 22.5.1 is released.

**Review by ljharb — APPROVED** [review]

**Pull request merged**